calico_containers/calico_ctl/profile.py

```diff
@@ -150,14 +153,27 @@ def profile_add(profile_name):
         print "Created profile %s" % profile_name
 
 
-def profile_remove(profile_name):
-    # TODO - Don't allow removing a profile that has endpoints in it.
-    try:
-        client.remove_profile(profile_name)
-    except KeyError:
-        print "Couldn't find profile with name %s" % profile_name
+def profile_remove(profile_name, nocheck=None):
+    """
+    Remove a profile as long as it does not contain any endpoints.
+    Allow user to explicitly remove the profile if desired.
+    :param profile_name: The name of the profile to remove.
+    :param nocheck: Flag saying to remove profile regardless of endpoints.
+    :return: None.
+    """
+    # Check if the profile exists.
+    if client.profile_exists(profile_name):
+        members = client.get_profile_members(profile_name)
```

Comment:
IIRC, getting the profile members is a fairly intensive operation - it has to do a full dump of the endpoints and then check if the profile is in the endpoint.

I don't think we should do that if the nocheck flag is set.